chia/full_node/block_store.py

```diff
@@ -155,15 +155,16 @@ async def rollback(self, height: int) -> None:
         if self.db_wrapper.db_version == 2:
             async with self.db_wrapper.writer_maybe_transaction() as conn:
                 await conn.execute(
-                    "UPDATE OR FAIL full_blocks SET in_main_chain=0 WHERE height>? AND in_main_chain=1", (height,)
+                    "UPDATE full_blocks SET in_main_chain=0 WHERE height>? AND in_main_chain=1", (height,)
                 )
 
     async def set_in_chain(self, header_hashes: List[Tuple[bytes32]]) -> None:
         if self.db_wrapper.db_version == 2:
             async with self.db_wrapper.writer_maybe_transaction() as conn:
-                await conn.executemany(
-                    "UPDATE OR FAIL full_blocks SET in_main_chain=1 WHERE header_hash=?", header_hashes
-                )
+                async with await conn.executemany(
+                    "UPDATE full_blocks SET in_main_chain=1 WHERE header_hash=?", header_hashes
+                ) as cursor:
+                    assert cursor.rowcount > 0
```

Comment:
Asserts should have a comment expressing what must have happened and why this shouldn't be possible and what sort of bug might trigger this.  Or some part of that. 
 Remember that the CLI result for relevant asserts is `{"success": False, "error": ""}` when there's no message.  Tracebacks from pyinstallered code don't even contain the code.  So you end up having to verify the version that was running and pull up that version in code to find what was even asserted.

Side note, I should find a linter setting to require this...  Albeit as a stop-gap until I get my way and get to ban `assert` entirely in runtime code.  `:]`

Reply:
I changed this to a `RuntimeError` with an informative error message